packages/medusa/src/services/event-bus.ts

```diff
@@ -332,18 +332,17 @@ export default class EventBusService {
       )
       const jobs = await stagedJobRepo.find(listConfig)
 
-      await Promise.all(
-        jobs.map((job) => {
-          this.queue_
-            .add(
-              { eventName: job.event_name, data: job.data },
-              { jobId: job.id, ...job.options }
-            )
-            .then(async () => {
-              await stagedJobRepo.remove(job)
-            })
-        })
-      )
+      const eventsData = jobs.map((job) => {
+        return {
+          eventName: job.event_name,
+          data: job.data,
+          opts: { jobId: job.id, ...job.options },
+        }
+      })
+
+      await this.queue_.addBulk(eventsData).then(async () => {
+        return await stagedJobRepo.delete({ id: In(jobs.map((j) => j.id)) })
+      })
```

Comment:
praise: I think this is a very neat implementation!

Reply:
@adrien2p I see this might have tampered with the batch job integration tests. Can I get you to investigate? Would be great to get this merged soon :)

Reply:
Ahah yes i am on it already